eth_crowdfund_be/eth_crowdfund_api/src/tests/test_campaign.py

```diff
@@ -0,0 +1,155 @@
+    # just checking a sad path
+    self.assertNotEqual(self.campaign1["min_contribution"], 100)
+    self.assertEqual(res.status_code, 201)
+
+  def test_campaign_update(self):
+    campaign1_update = {
+      "description": "new description", 
+      "contributors": 5,
+      "manager": "1x3y"
+    }
+    # create the user before updating
+    res = self.client().post('/api/v1/campaigns/',
+                             headers={'Content-Type': 'application/json'},
+                             data=json.dumps(self.campaign1))
+    self.assertEqual(res.status_code, 201)
+    campaign1_id = json.loads(res.data)["id"]
+    # save the id and update user
+    res_update = self.client().put('/api/v1/campaigns/{}'.format(campaign1_id), 
+                            headers={'Content-Type': 'application/json'},
+                            data=json.dumps(campaign1_update))
+
+    json_data = json.loads(res_update.data)
+    self.assertEqual(self.campaign1["name"], json_data["name"])
+    self.assertNotEqual(self.campaign1["description"], json_data["description"])
+    self.assertEqual(campaign1_update["description"], json_data["description"])
+    self.assertNotEqual(self.campaign1["contributors"], json_data["contributors"])
+    self.assertEqual(campaign1_update["contributors"], json_data["contributors"])
+    self.assertNotEqual(self.campaign1["manager"], json_data["manager"])
+    self.assertEqual(campaign1_update["manager"], json_data["manager"])
+    self.assertEqual(res_update.status_code, 200)
+  
+  def test_delete(self):
+    res = self.client().post('/api/v1/campaigns/',
+                              headers={'Content-Type': 'application/json'},
+                              data=json.dumps(self.campaign1))
+    self.assertEqual(res.status_code, 201)
+    campaign1_id = json.loads(res.data)["id"]
+
+    # delete campaign
+    res_delete = self.client().delete('/api/v1/campaigns/{}'.format(campaign1_id), headers={'Content-Type': 'application/json'})
+    self.assertEqual(res_delete.status_code, 200)
+  
+  def test_get_all_campaigns(self):
+    # create 2 campaigns
+    res1 = self.client().post('/api/v1/campaigns/',
+                             headers={'Content-Type': 'application/json'},
+                             data=json.dumps(self.campaign1))
+    self.assertEqual(res1.status_code, 201)
+    res2 = self.client().post('/api/v1/campaigns/',
+                             headers={'Content-Type': 'application/json'},
+                             data=json.dumps(self.campaign2))
+    self.assertEqual(res2.status_code, 201)
+
+
+    res_index = self.client().get('/api/v1/campaigns/',
+                   headers={'Content-Type': 'application/json'})
+    json_data = json.loads(res_index.data)
+    self.assertEqual(len(json_data), 2)
+    self.assertEqual(res_index.status_code, 200)
+
+  # def test_get_one_campaign(self):
+  #   # create 2 campaigns
```

Comment:
Not sure why this test wouldnt pass. When sending a request to `get('/api/v1/campaigns/<campaign_id>...` I was getting a status code of 308 which is a `permanent redirect message`

Reply:
Did you try `get('/api/v1/campaigns/<campaign_id>/`? Sometimes I have success adding the final `/` when testing.